source/core/rtc_adapter/RtcAdapter.cc

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) <2020> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <RtcAdapter.h>
+#include <AdapterInternalDefinitions.h>
+#include <VideoReceiveAdapter.h>
+#include <VideoSendAdapter.h>
+// #include <AudioReceiveAdapter.h>
+#include <AudioSendAdapter.h>
+
+#include <memory>
+
+namespace rtc_adapter {
+
+class RtcAdapterImpl : public RtcAdapter,
+                       public CallOwner {
+public:
+    RtcAdapterImpl();
+    virtual ~RtcAdapterImpl();
+
+    // Implement RtcAdapter
```

Comment:
This is not good practice to have so many create/destroy functions that's public. Can you consolidate it?